Modules/_sqlite/cursor.c

```diff
@@ -121,6 +117,10 @@ cursor_dealloc(pysqlite_Cursor *self)
     if (self->in_weakreflist != NULL) {
         PyObject_ClearWeakRefs((PyObject*)self);
     }
+    /* Reset the statement if the user has not closed the cursor */
+    if (self->statement) {
```

Comment:
This will always be NULL if the GC has collected the object in a cycle because you called `Py_CLEAR` in `tp_clear`.

Reply:
Correct. Thanks.

Reply:
Maybe I'm missing something: `tp_clear` is called _after_ statement reset in `cursor_dealloc`.

Reply:
> Maybe I'm missing something: `tp_clear` is called _after_ statement reset in `cursor_dealloc`.

The GC calls `tp_clear` *before* the objects gets deallocated

Reply:
So the explicit `tp_clear` in `cursor_dealloc` is just for corner cases?